utils/utils.go

```diff
@@ -57,12 +58,34 @@ func DetermineNodename(conf types.NetConf) string {
 		nodename = conf.Hostname
 		logrus.Warn("Configuration option 'hostname' is deprecated, use 'nodename' instead.")
 	}
+	if nff := nodenameFromFile(); nff != "" {
+		logrus.Debugf("Read node name from file: %s", nff)
+		nodename = nff
+	}
 	if conf.Nodename != "" {
```

Comment:
Should this be an `else if`? If multiple methods are provided (albeit an unusual edge case), the lowest priority would take precedence.

Reply:
The way the code is structured now is acting as designed - the `conf.Nodename` takes priority over the `/var/lib/calico/nodename` file.